tools/mo/openvino/tools/mo/back/ResultRename.py

```diff
@@ -13,18 +13,33 @@ class ResultRename(BackReplacementPattern):
     enabled = False
 
     def find_and_replace_pattern(self, graph: Graph):
+        names_set = set()
+        for node in graph.get_op_nodes():
+            if node.has_valid('name'):
+                names_set.add(node['name'])
+
         for node in graph.get_op_nodes(type='Result'):
             if node.in_ports():
                 prev_node_out_port = node.in_port(0).get_connection().get_source()
                 tensor_names = prev_node_out_port.get_tensor_names()
                 # Graph may contain Result nodes with names equal to input tensors and
                 # renaming in this case is not needed. The example of such situation is
                 # IR reader check when graph is read with correct Result names.
-                if tensor_names and node.soft_get('name') == tensor_names[0]:
+                if not tensor_names:
+                    continue
+                # If Result name is equal to some tensor name from list, then renaming is not needed
+                if node.soft_get('name') in tensor_names:
                     continue
-                if tensor_names and not graph.get_op_nodes(name=tensor_names[0]):
-                    result_name = tensor_names[0]
-                else:
+
+                # Try to find tensor name, that is not intersects with graph node names
+                result_name = None
+                for tensor_name in tensor_names:
+                    if tensor_name not in names_set:
+                        result_name = tensor_name
```

Comment:
since you find a new name for `Result` node you need to remove old name from `names_set` and add a new one

Reply:
Done.